indra/llcommon/lua_function.cpp

```diff
@@ -68,6 +68,15 @@ int lluau::loadstring(lua_State *L, const std::string &desc, const std::string &
     return luau_load(L, desc.data(), bytecode.get(), bytecodeSize, 0);
 }
 
+std::filesystem::path lluau::source_path(lua_State* L)
+{
+    //Luau lua_Debug and lua_getinfo() are different compared to default Lua:
+    //see https://github.com/luau-lang/luau/blob/80928acb92d1e4b6db16bada6d21b1fb6fa66265/VM/include/lua.h
+    lua_Debug ar;
+    lua_getinfo(L, 1, "s", &ar);
```

Comment:
No error checking or pointer verification (L could be nullptr) makes me sad

Reply:
You have a good point, but this all agrees with embedded Lua convention: nobody expects ~~the Spanish~~ a `nullptr` `lua_State*`. I could defend this whole source file against `(! L)` (by skipping the whole function body? by aborting the viewer?) but I suspect that if invalid `lua_State*` pointers start flying around, we'd just crash in the Lua VM anyway.